crates/matrix-sdk/src/oidc.rs

```diff
@@ -709,25 +703,40 @@ impl OidcLoginBuilder {
     ///
     /// [`ClientErrorCode`]: matrix_sdk::oidc::types::errors::ClientErrorCode
     #[instrument(target = "matrix_sdk::client", skip_all)]
-    pub async fn login_with_authorization_code<F, Fut>(
+    pub async fn url_for_login_with_authorization_code(
         self,
         redirect_uri: &Url,
-        use_authorization_url: F,
-    ) -> Result<AccessTokenResponse, OidcError>
-    where
-        F: FnOnce(Url, String) -> Fut + Send,
-        Fut: Future<Output = Result<String, OidcError>> + Send,
-    {
+    ) -> Result<OidcAuthorizationData, OidcError> {
         let issuer = self.oidc.issuer().await.ok_or(OidcError::MissingAuthenticationIssuer)?;
         info!(issuer, "Logging in via OpenID Connect with the Authorization Code flow");
 
         let scope = generate_scope(self.device_id)?;
 
-        let response = self
-            .oidc
-            .authorize_scope_with_authorization_code(&scope, redirect_uri, use_authorization_url)
-            .await?;
+        self.oidc.url_for_scope_with_authorization_code(&scope, redirect_uri).await
+    }
 
+    /// Finish logging in via the Authorization Code flow.
+    ///
+    /// This method should be called after the URL returned by
+    /// [`url_for_login_with_authorization_code`] has been presented and
+    /// has redirected the user to to the redirect URI.
+    ///
+    /// # Arguments
+    ///
+    /// * `code` - The code received as part of the redirect URI when login was
+    ///   successful.
+    ///
+    /// * `state` - The unique string received as part of the redirect URI when
+    ///   login was successful
+    ///
+    /// [`ClientErrorCode`]: matrix_sdk::oidc::types::errors::ClientErrorCode
+    #[instrument(target = "matrix_sdk::client", skip_all)]
+    pub async fn login_with_authorization_code(
```

Comment:
Didn't I leave a comment to rename this?

```suggestion
    pub async fn finish_login_with_authorization_code(
```

Reply:
Oh, hmmm. I didn't see that one, but I updated the docs for it.